src/components/navigation-header/components/dropdown-menu/index.tsx

```diff
@@ -0,0 +1,190 @@
+ * menu item groups.
+ *
+ * TODO: add more details as more interaction support is added
+ */
+const NavigationHeaderDropdownMenu = ({
+  label,
+  itemGroups,
+}: NavigationHeaderDropdownMenuProps) => {
+  const menuRef = useRef<HTMLDivElement>()
+  const activatorButtonRef = useRef<HTMLButtonElement>()
+  const [isOpen, setIsOpen] = useState(false)
+  const numberOfItemGroups = itemGroups.length
+  const menuId = useMemo(() => `menu-${slugify(label)}`, [label])
+
+  // Handles closing the menu if there is a click outside of it and it is open.
+  useOnClickOutside([menuRef], () => setIsOpen(false), isOpen)
+
+  // Handles closing the menu if focus moves outside of it and it is open.
+  useOnFocusOutside(menuRef, () => setIsOpen(false), isOpen)
+
+  /**
+   * Generates a unique ID for a single dropdown menu item based on the ID of
+   * the group it belongs to.
+   */
+  const getItemId = (groupId: string, itemIndex: number): string => {
+    return `${groupId}-item-${itemIndex}`
+  }
+
+  /**
+   * Generates a unique ID for a group of items based on the main menu ID and
+   * the index of the group.
+   */
+  const getItemGroupId = (groupIndex: number): string => {
+    return `${menuId}-itemGroup-${groupIndex}`
+  }
+
+  /**
+   * Handles click interaction with the activator button. When clicked, if the
+   * menu is:
+   *  - open, then it will be closed
+   *  - closed, then it will be opened
+   */
+  const handleClick = () => {
+    setIsOpen(!isOpen)
+  }
+
+  /**
+   * Handles the behavior that should happen when a key is pressed down.
+   * Currently used by both the activator button and each menu item anchor
+   * element. Currently only handles what happens when the Escape is pressed
+   * because all other keyboard interaction is handled by default interactions
+   * with these elements.
+   *
+   * On Escape:
+   *  - the menu is closed, if it is open
+   *  - the activator button is given focus
+   */
+  const handleKeyDown = (e: KeyboardEvent) => {
+    const { isEscapeKey } = deriveKeyEventState(e)
+    if (isEscapeKey) {
+      setIsOpen(false)
+      activatorButtonRef.current.focus()
+    }
+  }
+
+  /**
+   * Handles the start of a mouse hover interaction with the activator button.
+   * When the mouse pointer hovers over the activator button, the menu will be
+   * opened if it is not already open.
+   */
+  const handleMouseEnter = () => {
+    if (!isOpen) {
+      setIsOpen(true)
+    }
+  }
+
+  /**
+   * Handles the end of a mouse hover interaction with the entire menu. If the
+   * menu is open, and the mouse moves outside the bounds either the activator
+   * button or the dropdown menu list, then the menu will be closed.
+   */
+  const handleMouseLeave = () => {
+    if (isOpen) {
+      setIsOpen(false)
+    }
+  }
+
+  return (
+    <div className={s.root} onMouseLeave={handleMouseLeave} ref={menuRef}>
+      <div className={s.activatorWrapper}>
+        <button
```

Comment:
Looks to be some unexpected behavior with regards to keyboard navigation happening in Firefox. Unable to tab into the menu items.

https://user-images.githubusercontent.com/825855/161990495-301a7869-921d-4f03-ac5c-b833df7b3c9d.mp4

![image](https://user-images.githubusercontent.com/825855/161990597-42505d90-2bd4-481e-9eb3-b339c00d6cf3.png)

Reply:
I am able to tab to menu items in Firefox. Can you see if you have `accessibility.tabfocus` in your config as described here: https://www.a11yproject.com/posts/macos-browser-keyboard-navigation/#highlight-each-item-in-firefox.

When I have that configured and set to 7 as that section describes, I cannot TAB to links in the dropdown. When I delete the setting, I am then able to TAB to the links. It's possible that this is no longer a needed setting? [Scott O'Hara has a post with a 11/13/2019 update](https://www.scottohara.me/blog/2014/10/03/link-tabbing-firefox-osx.html#latest_update) that says: "However, apparently in Firefox 71+, an update will be put in place where this may no longer be necessary"

Reply:
I do have it enabled 
![CleanShot 2022-04-06 at 11 19 13@2x](https://user-images.githubusercontent.com/825855/162009319-c247cde8-9a17-43d9-afda-4e6e7259ca93.png)

Reply:
Could you delete it and see if that fixes your issue?

Reply:
Still wondering why i'd be able to tab to the top level buttons but not the menu items without the setting correctly in place. I'd assume I wouldn't have been able to tab through the trigger elements. 🤷🏼